### PR TITLE
Add nix linter

### PR DESCRIPTION
**Pull request by gpanders**
Add a linter for `nix` filetypes.

### CODE REVIEW

**Review by mfussenegger — COMMENTED** [review]
Thanks for the PR, could you rebase this against latest master and use the newly improved from_pattern function? See #41

**Comment by gpanders** [comment]
Sure, done. The new `from_pattern` method is nifty.

Note that I also added one other commit to this PR to address my comment [here](https://github.com/mfussenegger/nvim-lint/pull/41/files#r672673259). This allows patterns to use `^` as a start-of-line anchor.

File: README.md [hunk 1]
```diff
@@ -80,6 +80,7 @@ Other dedicated linters that are built-in are:
 | [luacheck][19]      | `luacheck`     |
 | [chktex][20]        | `chktex`       |
 | [vint][21]          | `vint`         |
+| nix                 | `nix`          |
```
Comment (Tastyep) [inline README.md:83]: Can you link this to the [Github page](https://github.com/Synthetica9/nix-linter) ?

Comment (gpanders) [reply README.md:83]: This does not use `nix-linter`, it uses `nix-instantiate --parse` which is built-in to Nix.

Comment (Tastyep) [reply README.md:83]: I didn't check, sorry about that

**Comment by mfussenegger** [comment]
Thanks for the PR :+1:

**Pull request merged**